apps/sms/views/inbox/js/inbox.js

```diff
@@ -674,13 +606,16 @@ var InboxView = {
         this.startRendering();
       }
 
-      this.appendThread(thread);
-
-      // Dispatch visually-loaded when rendered threads could fill up the top of
-      // the visible area.
-      if (--firstPanelCount === 0) {
-        this.emit('visually-loaded');
-      }
+      this.appendThread(thread).then(() => {
+        // Dispatch visually-loaded when rendered threads could fill up the top
+        // of the visible area.
+        if (--firstPanelCount === 0) {
+          this.list.setModel(this.listData);
+          this.emit('visually-loaded');
+        } else if (this.listData.length % this.FIRST_PANEL_THREAD_COUNT === 0) {
+          this.list.setModel(this.listData);
```

Comment:
yes because I want to reduce the call of setModel for rendering. Maybe we can compare with  item-by-item rendering to see if calling setModel frequently does bring obvious side effects.